### PR TITLE
refactor: lease client api 

### PR DESCRIPTION
**Pull request by LingKa28**
Base on #16 

Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)
Refactor lease client api to improve easy of use of the lease clinet feature.

* what changes does this pull request make?
  * refactor lease clinet api & test sample

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)
No.

### CODE REVIEW

File: client/lease.go [hunk 1]
```diff
@@ -43,73 +65,104 @@ func newLeaseClient(
...
 
 	res, err := c.curpClient.propose(&cmd, true)
 	if err != nil {
 		return nil, err
 	}
-	return res.CommandResp.GetLeaseRevokeResponse(), err
+	return (*LeaseRevokeResponse)(res.CommandResp.GetLeaseRevokeResponse()), err
 }
 
+// TODO:
```
Comment (Phoenix500526) [inline client/lease.go:107]: todo what？

Comment (LingKa28) [reply client/lease.go]: This feature is currently being done.

**Review by Phoenix500526 — APPROVED** [review]
LGTM

File: client/lease.go [hunk 2]
```diff
@@ -34,82 +84,141 @@ func newLeaseClient(
...
+				if err != nil {
+					c.logger.Error("keep alive fail", zap.Error(err))
+				}
+				res, err := stream.Recv()
+				if err != nil {
+					c.logger.Error("keep alive fail", zap.Error(err))
+				}
+				kach <- (*LeaseKeepAliveResponse)(res)
+			}
+			time.Sleep(1 * time.Millisecond)
```
Comment (iGxnon) [inline client/lease.go:162]: Could we make it configurable or a constant?

Comment (LingKa28) [reply client/lease.go:162]: I changed it to a constant.

**Review by Phoenix500526 — COMMENTED** [review]
Please modify the commit messages and make it brief and clear. For instances, the commit message of the second commit, "test: refactor lease client api", what exactly does it convey - adding tests or refactoring?

**Comment by LingKa28** [comment]
> Please modify the commit messages and make it brief and clear. For instances, the commit message of the second commit, "test: refactor lease client api", what exactly does it convey - adding tests or refactoring?

I've changed it.

**Review by Phoenix500526 — APPROVED** [review]
LGTM

**Review by iGxnon — APPROVED** [review]
LGTM

**Pull request merged**